src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -871,12 +871,21 @@ private void EmitFieldLoad(BoundFieldAccess fieldAccess, bool used)
         {
             var field = fieldAccess.FieldSymbol;
 
-            //TODO: For static field access this may require ..ctor to run. Is this a side-effect?
-            // Accessing unused instance field on a struct is a noop. Just emit the receiver.
-            if (!used && !field.IsVolatile && !field.IsStatic && fieldAccess.ReceiverOpt.Type.IsVerifierValue())
+            if (!used)
             {
-                EmitExpression(fieldAccess.ReceiverOpt, used: false);
-                return;
+                // fetching unused captured frame is a no-op (like reading "this")
+                if(field.IsCapturedFrame)
+                {
+                    return;
+                }
+
+                //TODO: For static field access this may require ..ctor to run. Is this a side-effect?
```

Comment:
Note that generally we cannot generally know whether we have .cctor or not, since the type maybe in a referenced assembly. 
In some cases we might know, but does not seem to be worth special casing.

---
In reply to: [87655591](https://github.com/dotnet/roslyn/pull/14959#discussion_r87655591) [](ancestors = 87655591,87618871)